File: BinaryGap/BinaryGap.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(n) {
  let binary = n.toString(2)
  let num = binary.match(/1/g).length; 
  if(num === 1){
    return 0
  }else{
    binary = binary.split('1')
    for (let i = 0; i < binary.length; i++) {
      binary[i] = binary[i].length
    }
    return Math.max(...binary)
  }
}

File: BinaryGap/BinaryGap.c
// you can write to stdout for debugging purposes, e.g.
// printf("this is a debug message\n");

int solution(int N) {
	int num = 0, count = 0, flag = 0, result = 0;

	do {
		num = N % 2;
		N /= 2;

		if (num == 0) {
			count++;
		}
		else {
			if (result < count) {
				flag++;
				result = count;
				count = 0;
			}
			else {
				count = 0;
				flag++;
			}
		}
	} while (N > 0);

	if (flag == 1)
		result = 0;

	printf("%d\n", result);
	return result;
}